### ISSUE TITLE
Java: Issue with enum parsing (xml output)

### ISSUE THREAD

**Issue by Stewori** [doxygen/doxygen#8925]
**Describe the bug**
Processing Java code
```
public enum Test {
    /**
     * doc 1.
     */
    VALUE1 {
        void method() {
            // some code...
        }
    },

    /**
     * doc 2.
     */
    VALUE2 {
        void method() {
            // some code...
        }
    };

    /**
     * doc abstract
     */
    abstract void method();
}
```
results in the following xml snippet:
```
...
    <compoundname>Test</compoundname>
      <sectiondef kind="public-attrib">
      <memberdef kind="variable" id="enumTest_1af5cfe57306c2ff48fdc5513176124b5b" prot="public" static="no" mutable="no">
        <type></type>
        <definition>Test.method</definition>
        <argsstring></argsstring>
        <name>method</name>
        <initializer>=() {
            
        }
    },

    
    VALUE2 {
        void method() {
            
        }
    }</initializer>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para>doc 1. </para>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Test.java" line="18" column="1" bodyfile="Test.java" bodystart="6" bodyend="-1"/>
      </memberdef>
      </sectiondef>
...
```
Neither ``VALUE1`` nor ``VALUE2`` is recognized as a field. Instead one of the inner ``method`` declarations is mistaken as a class-level member. The abstract method is not really required for this example, it is just for illustrative purpose. It is processed as expected.

**Expected behavior**
I expect that the resulting code snippet would be much like processing
```
public enum Test2 {
    /**
     * doc 1.
     */
    VALUE1,

    /**
     *doc 2.
     */
    VALUE2;
}
```

i.e.

```
    <compoundname>Test2</compoundname>
      <sectiondef kind="public-attrib">
      <memberdef kind="variable" id="enumTest2_1a6ecee2002080e657d8f95674ad4cdbbb" prot="public" static="no" mutable="no">
        <type></type>
        <definition>Test2.VALUE1</definition>
        <argsstring></argsstring>
        <name>VALUE1</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para>doc 1. </para>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Test2.java" line="6" column="1" bodyfile="Test2.java" bodystart="6" bodyend="-1"/>
      </memberdef>
      <memberdef kind="variable" id="enumTest2_1a0af1d89737f05e04e8bf32e7038cb60d" prot="public" static="no" mutable="no">
        <type></type>
        <definition>Test2.VALUE2</definition>
        <argsstring></argsstring>
        <name>VALUE2</name>
        <briefdescription>
        </briefdescription>
        <detaileddescription>
<para>doc 2. </para>
        </detaileddescription>
        <inbodydescription>
        </inbodydescription>
        <location file="Test2.java" line="11" column="1" bodyfile="Test2.java" bodystart="11" bodyend="-1"/>
      </memberdef>
      </sectiondef>
```
Probably, for the complex example ``Test.java``, the fields should each contain an initializer as follows
```
<initializer>{
        void method() {
            
        }
    }</initializer>
```

**To Reproduce**
Find a Java file and doxygen config attached. The issue is demonstrated like mentioned above.
After running doxygen on the example, the files in the subfolder ``xml`` contain the snippet shown above.
[issue_enum.zip](https://github.com/doxygen/doxygen/files/7626742/issue_enum.zip)

**Version**
Tested with doxygen clone from 2021-11-08:
```
/blah/doxygen/build/bin/doxygen -v
1.9.3 (096a1c10e1e0b86edc8c561f49b5fd7881776ba7)
```

`lsb_release -a`:
```
LSB Version:	core-9.20160110ubuntu0.2-amd64:core-9.20160110ubuntu0.2-noarch:security-9.20160110ubuntu0.2-amd64:security-9.20160110ubuntu0.2-noarch
Distributor ID:	LinuxMint
Description:	Linux Mint 18 Sarah
Release:	18
Codename:	sarah
```

**Additional context**
TBH I did not check for duplicates. Please excuse if this is a known issue.

**Comment by albert-github** [comment]
I think the main issue here is the discrepancy between the C++ and Java forms of the enum.
In C++ an enum consists just of values like also shown in the second test file (`Test2.java`).
In the Java version we have:
- constants with some block of code behind it
- some functions after the constants (from the documentation "In addition to enum constants, the body of an enum declaration may contain constructor and member declarations as well as instance and static initializers.")

The later seems to be handled properly, the problem is caused by the "code block" behind the constant.

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
